improver/cli/apply_night_mask.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# (C) British Crown Copyright 2017-2020 Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Script to set night values to zero within the input cube."""
+
+from improver import cli
+
+
+@cli.clizefy
+@cli.with_output
+def process(cube: cli.inputcube):
+    """Masks night values from input cube to generate day only output.
+
+    Args:
+        cube (iris.cube.Cube):
+            Cube to be masked.
+
+    Returns:
+        iris.cube.Cube:
+            Input cube with all night values masked out.
+
+    """
+
+    import numpy as np
+    import numpy.ma as ma
+
+    from improver.utilities.solar import DayNightMask
+
+    mask = DayNightMask()(cube).data
+    # Broadcast mask to shape of input cube to account for additional dimensions.
+    mask = np.broadcast_to(mask, cube.shape)
+    # setting night values to zero.
+    cube.data = ma.where(mask == DayNightMask().night, 0, cube.data)
```

Comment:
Is there a reason to use `ma.where` over `np.where`? `np.where` is ~twice as fast, and the mask in the resulting array is set to false here anyway.

Reply:
Yes, using `np.where` rather than `ma.where` is a good idea.

You're correct that this CLI could be easily extended to apply a day mask, however, as we don't currently have a use case for a day mask, I would be reluctant to add code to the codebase that we don't have a use case for, particularly as we often spend time stripping out unused and unnecessary code and stripping out unused CLI options. Therefore, I think the current preference would be to wait until / if a use case arises for a day mask utility, and then extend this CLI then.

Reply:
I think stripping out unused CLI options is a good idea if it reduces the code to maintain, but in this case, it would be as simple as an extra parameter fed into the function. My feeling is always to make the code more general, but I see where you're coming from if there is currently no use for it.

Just having a quick look at the JIRA ticket, I can see a potential issue has been highlighted with setting all the values to 0, can I ask why a mask isn't used here? i.e. using something like `ma.array(cube.data, mask=mask==DayNightMask().night)`

Reply:
I think the ticket is slightly out of date. @gavinevans is going to double check exactly what the output should look like. I think question about whether the generalisation makes sense right now or not will depend on what the output should look like. Once Gavin has checked the output we should be able to pin down answers to these questions

Reply:
Okay, thanks @fionaRust

Reply:
I've now confirmed with @PaulAbernethy that the UV index for the night values needs to be set to zero to match our existing UV index output, so that the final output is in the format that is expected by our users. More information is on the JIRA ticket. Therefore this CLI needs to be specifically aimed at the requirement of setting night values to a UV index of zero.

Reply:
Okay, thanks @gavinevans. So just the switch from `ma.where` to `np.where` from me then :-)